examples/protogen1.html

```diff
@@ -0,0 +1,480 @@
+			<spacedlayout spacing="8" updateparent="true"></spacedlayout>
+			<slider name="hslide" y="5" width="250" height="30" value="50" minvalue="15" maxvalue="250" bgcolor="#808080">
+			<handler event="onmouseup">
+				theboard.boardwidthmm = Math.round(this.parent.hslide.value);
+				</handler>
+				</slider>		
+			<text  color="white" text="${Math.round(this.parent.hslide.value)}" y="${this.parent.hslide.y + (this.parent.hslide.height-this.height)/2}"></text>
+			</view>
+		<text text="height in MM" color="white"></text>		
+			<view width="100%" height="auto">
+			<spacedlayout spacing="8" updateparent="true"></spacedlayout>
+			<slider name="hslide" y="5" width="250" height="30" value="50" minvalue="15" maxvalue="250" bgcolor="#808080">
+			<handler event="onmouseup">
+				theboard.boardheightmm = Math.round(this.parent.hslide.value);
+				</handler>
+			</slider>		
+			<text  color="white" text="${Math.round(this.parent.hslide.value)}" y="${this.parent.hslide.y + (this.parent.hslide.height-this.height)/2}"></text>
+		</view>
+		<text text="holes per segment"  color="white"></text>		
+		<view width="100%" height="auto" >
+			<spacedlayout spacing="8" updateparent="true"></spacedlayout>
+			<slider name="hslide" y="5" width="250" height="30" value="5" minvalue="2" maxvalue="10" bgcolor="#808080">
+				<handler event="onmouseup">
+				theboard.holespersegment = Math.round(this.parent.hslide.value);
+				</handler>
+			</slider>		
+			<text  color="white" text="${Math.round(this.parent.hslide.value)}" y="${this.parent.hslide.y + (this.parent.hslide.height-this.height)/2}"></text>
+			</view>
+			<text  color="white" text="segments between rails"></text>	
+			<view width="100%" height="auto" name="segments">
+				<spacedlayout spacing="8" updateparent="true"></spacedlayout>
+				<slider name="hslide" y="5" width="250" height="30" value="2" minvalue="1" maxvalue="10" bgcolor="#808080">
+					<handler event="onmouseup">
+				theboard.segmentsbetweenrails = Math.round(this.parent.hslide.value);
+				theboard.updateall();
+				</handler>
+				
+				
+				</slider>		
+				<text  color="white" text="${Math.round(this.parent.hslide.value)}" y="${this.parent.hslide.y + (this.parent.hslide.height-this.height)/2}"></text>
+			</view>
+			<checkbutton text="Siderails" color="white" selectcolor="green" defaultcolor="red"  selected="true">
+				<handler event="onselected" args="value">
+					theboard.siderails = value;
+				</handler>
+			</checkbutton>
+			<checkbutton text="Gnd and VCC rails" color="white" selectcolor="green" defaultcolor="red" selected="true">
+				<handler event="onselected" args="value">
+					theboard.gndvcclines = value;
+				</handler>
+			</checkbutton>
+			<zbutton text="Regenerate board">
+			<handler event="onclick">
+				theboard.updateall();
+			</handler>
+			</zbutton>
+			
+		</view>
+		</view>
+		<view name="blah2" height="100%" width="${this.parent.width - (dragger.x+10)}" x ="${dragger.x+10}" bgcolor="lightgray">

```

Comment:
Aww man - this could use some prettification, same with the JS below! No biggie tho.